src/lib/ptree/ptree.lua

```diff
@@ -347,25 +347,34 @@ function Manager:monitor_worker_counters(id)
    local events = inotify.recursive_directory_inventory_events(dir, cancel)
    for ev in events.get, events do
       if has_suffix(ev.name, '.counter') then
-         local stat = S.lstat(ev.name)
          local name = strip_prefix(ev.name, dir..'/')
          local qualified_name = '/'..pid..'/'..name
          local counters = self.counters[name]
-         if blacklisted(name) or not stat or stat.islnk then
+         if blacklisted(name) then
             -- Pass.
          elseif ev.kind == 'creat' then
             if not counters then
                counters = { aggregated=counter.create(name), active={},
                             rrd={}, aggregated_rrd=self:make_rrd(name),
-                            archived=ffi.new('uint64_t[1]') }
+                            archived=ffi.new('uint64_t[1]'),
+                            truename = {} }
                self.counters[name] = counters
             end
-            counters.active[pid] = counter.open(qualified_name)
-            counters.rrd[pid] = self:make_rrd(qualified_name)
+            local stat = S.lstat(ev.name)
+            if stat then
+               local truename = (stat.islnk and S.readlink(ev.name)) or ev.name
```

Comment:
I just checked and `S.readlink` returns nil on error -- so this works for me.  Great.